### PR TITLE
[Str] add grarpheme support

### LINKED ISSUE #62: [Str] Grapheme support
Just like `Psl\Str\Byte` component, we want to add `Psl\Str\Grapheme` that wraps grapheme_str* function in a nicer API that is consistent with our `Psl\Str` and `Psl\Str\Byte` components.

### PR DESCRIPTION
**Pull request by azjezz**
closes #62 
refs #63

### CODE REVIEW

File: src/Psl/Str/Grapheme/length.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+use function grapheme_strlen;
+
+/**
+ * Returns the length of the given string in grapheme units
+ *
+ * @psalm-pure
+ */
+function length(string $string): int
+{
+    return (int) grapheme_strlen($string);
```
Comment (azjezz) [inline src/Psl/Str/Grapheme/length.php]: I think there's a case where grapheme_strlen returns `null`, not sure how to handle that.

Comment (veewee) [reply src/Psl/Str/Grapheme/length.php]: The docs state it always returns an int though:
https://www.php.net/manual/en/function.grapheme-strlen.php

(We could keep it - just to be sure :) )

**Review by veewee — DISMISSED** [review]
Besides the conflicts, it Looks OK to me...

File: src/Psl/Str/Grapheme/length.php (hunk 1 above)
Comment (azjezz) [reply src/Psl/Str/Grapheme/length.php]: I learned to never belive the documentation 😆  

see:

- https://github.com/php/php-src/blob/e73bf01f9d0c23dd5baed4445b37c93ec4e0ab29/ext/intl/php_intl.stub.php#L246
- https://github.com/php/php-src/blob/7af24eaedd310911940643d7a5f412393b973389/ext/intl/grapheme/grapheme_string.c#L84
- https://github.com/php/php-src/blob/7af24eaedd310911940643d7a5f412393b973389/ext/intl/grapheme/grapheme_string.c#L96

not sure how to handle these cases ...

Comment (azjezz) [reply src/Psl/Str/Grapheme/length.php]: I added invariant violation checks.

**Pull request merged**